### PR TITLE
mark columns as not required if it has server_default field

### PR DESCRIPTION
**Pull request by asif-mahmud**
Check for `server_default` field in `Column` definition

### CODE REVIEW

**Comment by asif-mahmud** [comment]
Updated failed test code

**Comment by dkellner** [comment]
Thank you for your PR! Looks like py.test dropped Python 3.3 support (see https://docs.pytest.org/en/latest/changelog.html#pytest-3-3-0-2017-11-23), so we will do the same. I will take a look at that failing flake8 test.

**Comment by asif-mahmud** [comment]
And one more thing, for python3.5 , flake8 also complains about empty `except` in __init__.py.

File: .gitignore [hunk 1]
```diff
@@ -65,3 +66,6 @@ _build
 .idea
 
 .cache
+
+# vscode
+.vscode
```
Comment (dkellner) [inline .gitignore]: Please add a newline at the end of the file.

File: eve_sqlalchemy/config/fieldconfig.py [hunk 2]
```diff
@@ -56,11 +56,20 @@ def _render(self):
     def _get_field_nullable(self):
         return getattr(self._sqla_column, 'nullable', True)
 
+    def _has_field_server_default(self):
+        check = getattr(self._sqla_column, 'server_default')
+        if check:
+            return True
+        else:
+            return False
+
```
Comment (dkellner) [inline eve_sqlalchemy/config/fieldconfig.py]: This can be done simpler like this:
```python
return bool(getattr(self._sqla_column, 'server_default')
```

Comment (dkellner) [reply eve_sqlalchemy/config/fieldconfig.py]: Personally I would rename that method to just `has_server_default` as 'field' refers to the entry in Eve's resource configuration, not to the SQLAlchemy column.

File: eve_sqlalchemy/tests/config/resourceconfig/schema.py [hunk 3]
```diff
@@ -22,6 +22,8 @@ class SomeModel(Base):
     a_pickle = Column(types.PickleType)
     a_string = Column(types.String(42), default='H2G2')
     _internal = Column(types.Integer)
+    a_nullable_boolean = Column(types.Boolean, nullable=True)
+    a_server_default_col = Column(types.Integer, server_default='0')
```
Comment (dkellner) [inline eve_sqlalchemy/tests/config/resourceconfig/schema.py]: It looks strange to use an integer but pass a string as default; the SQLAlchemy docs use the `text` function here. Maybe just use `types.String` as the type doesn't really matter here.

File: eve_sqlalchemy/tests/config/resourceconfig/schema.py [hunk 4]
```diff
@@ -66,6 +69,10 @@ def test_required(self):
         # autoincrement='auto' per default (see SQLAlchemy docs for
         # details). As such, it is not required.
         self.assertFalse(schema['id']['required'])
+        # Columns with server_default field should be marked as not required
+        # additional test for nullable field set to True
+        self.assertFalse(schema['a_nullable_boolean']['required'])
+        self.assertFalse(schema['a_server_default_col']['required'])
```
Comment (dkellner) [inline eve_sqlalchemy/tests/config/resourceconfig/schema.py]: The comment does not provide additional information here, you can safely remove it.

**Pull request closed without merging**